### DISCUSSION TITLE
Set external Domain only for public shared links

### DISCUSSION THREAD

**Discussion by devtechk** [immich-app/immich#9231] [Show and tell]
Hello I would like to share my experience. I wanted to share only single photo or video without exposing the immich login page. I'm running Immich on a debian server with casaos. I've setup ngnix proxy manager (npm) on casaos. Everything in https. After setting up duckdns In casaos and opened the ngnix proxy manager port on the router, I've setup npm in this way
<img width="533" alt="Screenshot 2024-05-02 alle 21 44 53" src="https://github.com/immich-app/immich/assets/22073497/00010a77-4736-440f-b026-c5f286d27806">
<img width="505" alt="Screenshot 2024-05-02 alle 21 45 35" src="https://github.com/immich-app/immich/assets/22073497/19c31df3-c02c-468d-a17e-ec894bcf53dd">
and in the end I've setup the external domain in immich 
<img width="944" alt="Screenshot 2024-05-02 alle 21 34 44" src="https://github.com/immich-app/immich/assets/22073497/ae492d3e-c70d-4936-a0af-0a05784c0719">
I hope this is usefull for someone. 
Ask me for more.

**Comment by coolva** [comment]
Helpful? You solved days of frustrations with this short post. So, big thanks!

**Reply by devtechk** [reply]
Thank you very much! I also want to specify that "outside" I'm using a different port from internal lan sono ther's no problem at all. Before everything I forgot to explicit that I've installed the duckdns app from store to synch the ip :)

**Comment by JPBRM** [comment]
I've done something similar in NPM. But I've denied everything, except for every location that is needed (no more errors in de dev console of your browser).

In the `custom Nginx configuration` I've got the following (`192.168.178.179` is the local IP I'm using, substitute your own):

```
location / {
    return 403; 
}
location /share {
    proxy_pass http://192.168.178.179:2283;
}
location /_app {
    proxy_pass http://192.168.178.179:2283;
}
location /api {
    proxy_pass http://192.168.178.179:2283;
}
location /manifest.json {
    proxy_pass http://192.168.178.179:2283;
}
location /manifest-icon-192.maskable.png {
    proxy_pass http://192.168.178.179:2283;
}
location /custom.css {
    proxy_pass http://192.168.178.179:2283;
}
location /favicon.ico {
    proxy_pass http://192.168.178.179:2283;
}
```
